### PR TITLE
Added H5

### LINKED ISSUE #810: Request STM32H5 support!
The STM32 H5 is a new CortexM33 MCU from ST Micros.  ST released all the datasheets, reference manuals, updated cubemx a few days ago.

https://www.st.com/en/microcontrollers-microprocessors/stm32h5-series.html

https://www.st.com/en/microcontrollers-microprocessors/stm32h562rg.html

We need a PAC for it ASAP please!

I can't find the SVD files on ST's website yet, but using Eclipse CMSIS Packs plugin I was able to find a set of SVD's from Keil.  Attached.

[Keil-STM32H5-SVD-v1.0.0.zip](https://github.com/stm32-rs/stm32-rs/files/11009680/Keil-STM32H5-SVD-v1.0.0.zip)

### PR DESCRIPTION
**Pull request by David-OConnor**
Fixes #810

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -168,18 +170,28 @@ Check out the full list of supported devices [here](https://stm32-rs.github.io/s
...
 * You can run `scripts/matchperipherals.py` script to find out what existing
   peripherals could be cleanly applied to this new SVD. If they look sensible,
-  you can include them in your device YAML.
+  you can include them in your device YAML.  This requires a Python environment with the `pyyaml`
+  and `svdtools` dependencies.
 * Re-run `scripts/makecrates.py devices/` to update the crates with the new devices.
 * Run `make` to rebuild, which will make a patched SVD and then run `svd2rust`
   on it to generate the final library.
 
+### If adding a new STM32 family (not just a new device to an existing family), complete
```
Comment (adamgreig) [inline README.md:183]:
```suggestion
If adding a new STM32 family (not just a new device to an existing family), complete
```

**Comment by David-OConnor** [comment]
The basics are here; not sure how to locate which patches to apply. I attempted `matchperipherals.py`, but didn't find matches for any of the peripherals I tried. There are many things broken or inconsistent with existing PACs in this raw-SVD iteration.

File: README.md [hunk 2]
```diff
@@ -168,18 +170,28 @@ Check out the full list of supported devices [here](https://stm32-rs.github.io/s
 * Run `svd/extract.sh` to extract the zips into `svd` (ignored in git).
 * Add new YAML file in `devices/` with the new SVD path and include any
   required SVD patches for this device, such as renaming or merging fields.
+* Add the new devices to `stm32_part_table.yaml`.
+* Add the new devices `scripts/makecrates.py`.
 * You can run `scripts/matchperipherals.py` script to find out what existing
   peripherals could be cleanly applied to this new SVD. If they look sensible,
-  you can include them in your device YAML.
+  you can include them in your device YAML.  This requires a Python environment with the `pyyaml`
+  and `svdtools` dependencies. Example command: `python .\scripts\matchperipherals.py peripherals/rcc devices/stm32h562.yaml`
```
Comment (adamgreig) [inline README.md:178]:
```suggestion
  and `svdtools` dependencies.
  Example command: `python scripts/matchperipherals.py peripherals/rcc devices/stm32h562.yaml`
```

File: stm32_part_table.yaml [hunk 3]
```diff
@@ -343,6 +343,39 @@ stm32f7:
...
+  stm32h562:
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h5-series.html
+    rm: RM0481
+    rm_title: STM32H563/H573 and STM32H562 Arm®-based 32-bit MCUs
+    rm_url: https://www.st.com/resource/en/reference_manual/rm0481-stm32h563h573-and-stm32h562-armbased-32bit-mcus-stmicroelectronics.pdf
+    members:
+      - STM32H562
+
+  stm32h563:
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h5-series.html
```
Comment (adamgreig) [inline stm32_part_table.yaml:364]:
```suggestion
    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h563-573.html
```

File: stm32_part_table.yaml [hunk 4]
```diff
@@ -343,6 +343,39 @@ stm32f7:
...
+  stm32h503:
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h5-series.html
+    rm: RM0492
+    rm_title: STM32H503 line Arm®-based 32-bit MCUs
+    rm_url: https://www.st.com/resource/en/reference_manual/rm0492-stm32h503-line-armbased-32bit-mcus-stmicroelectronics.pdf
+    members:
+      - STM32H503
+
+  stm32h562:
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h5-series.html
```
Comment (adamgreig) [inline stm32_part_table.yaml:356]:
```suggestion
    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h562.html
```

File: stm32_part_table.yaml [hunk 5]
```diff
@@ -343,6 +343,39 @@ stm32f7:
       - STM32F778
       - STM32F779
 
+stm32h5:
+  stm32h503:
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h5-series.html
```
Comment (adamgreig) [inline stm32_part_table.yaml:348]:
```suggestion
    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h503.html
```

File: stm32_part_table.yaml [hunk 6]
```diff
@@ -343,6 +343,39 @@ stm32f7:
...
+  stm32h563:
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h5-series.html
+    rm: RM0481
+    rm_title: STM32H563/H573 and STM32H562 Arm®-based 32-bit MCUs
+    rm_url: https://www.st.com/resource/en/reference_manual/rm0481-stm32h563h573-and-stm32h562-armbased-32bit-mcus-stmicroelectronics.pdf
+    members:
+      - STM32H563
+
+  stm32h573:
+    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h5-series.html
```
Comment (adamgreig) [inline stm32_part_table.yaml:372]:
```suggestion
    url: https://www.st.com/en/microcontrollers-microprocessors/stm32h563-573.html
```

**Review by adamgreig — CHANGES_REQUESTED** [review]
Thanks for putting this together and updating the README too!

Just a few minor nits here. I also wondered if it would make sense to combine the 562/563/573 but it seems the 573 adds the entire security manager so probably not.

**Comment by David-OConnor** [comment]
Good stuff; I appreciate it!

**Review by adamgreig — APPROVED** [review]
thanks!

bors r+

**Comment by hydra** [comment]
Just trying this out and it seems that most of the registers and register blocks that I've looked have have the peripheral name prefixed to them.

e.g.

H5:
https://github.com/stm32-rs/stm32-rs-nightlies/tree/master/stm32h5/src/stm32h563/dbgmcu

vs. H7:
https://github.com/stm32-rs/stm32-rs-nightlies/tree/master/stm32h7/src/stm32h743/dbgmcu

H5 screenshot:

**Comment by joaomorenorf** [comment]
Hello,
I wasn't able to compile. Do you have a example project?
Best regards,
João

**Comment by David-OConnor** [comment]
Hi. The issue is most likely that there is no published H5 PAC. I may publish an interim one, then do a new release of this HAL to make it work.

**Pull request merged**